cmd/podman/machine/list.go

```diff
@@ -152,11 +166,12 @@ func streamName(imageStream string) string {
 	return imageStream
 }
 
-func toMachineFormat(vms []*machine.ListResponse, defaultCon string) []*entities.ListReporter {
+func toMachineFormat(vms []*machine.ListResponse, defaultCon *config.Connection) []*entities.ListReporter {
 	machineResponses := make([]*entities.ListReporter, 0, len(vms))
 	for _, vm := range vms {
+		isDefault := vm.Name == defaultCon.Name && strings.Contains(defaultCon.URI, strconv.Itoa((vm.Port)))
```

Comment:
I am not sure what the ssh check would add here? The can only be ever one connection with one name so it is impossible to create two VMs with the same name because the connection creation will fail.

Reply:
This fixes a bug, if we have two machines named the same thing on different providers, it will mark both machines as "default", because before this fix, we only checked if the connection name was the same as the machine name.

Reply:
I got that part but isn't it impossible to create two VMs with the same because as I said the connection name will cause a conflict failing the second creation. So unless a user manually modifies connections I see no way to get into that state.

Or to turn the logic around if two machine use the same connection name but different SSH ports how is the connection supposed to work? It could only ever work for one of them.

Reply:
Ah fair, I guess I was playing with connections. We could reach this state by doing a `podman system connection rm` and then creating another machine. Do you think it's worth preventing this very specific edge case, or is it better to just remove this logic and leave it as before?

Reply:
If you like to keep it I would not be against it but please add a comment that this is to ensure the connection matches the VM config.
However you will need to take care of the nil pointer deref pointed out above.